jcl/src/java.base/share/classes/java/lang/String.java

```diff
@@ -7740,9 +7747,16 @@ public String[] split(String regex, int max) {
 			} else {
 				int rLength = regex.lengthInternal();
 
-				char[] splitChars = regex.value;
+				char[] splitChars;
+				// if regex is compressed and this string isn't, decompress regex string
+				if (COMPACT_STRINGS && !compressed && (LATIN1 == regex.coder)) {
+					splitChars = new char[rLength << 1];
```

Comment:
This array should just have `rLength` elements.

Reply:
Changed to use `rLength` for size.